### ISSUE TITLE
v7 to 8.1 content migration fails due to missing data on a macro

### ISSUE THREAD

**Issue by nul800sebastiaan** [umbraco/Umbraco-CMS#5827]
The error is:

```
The database failed to upgrade. ERROR: The database configuration failed with the following message: Cannot insert the value NULL into column 'macroSource', table 'C:\GIT\SWAT-AUTO-UMBRACO8\APP_DATA\UMBRACO.MDF.dbo.cmsMacro'; column does not allow nulls. UPDATE fails.
 The statement has been terminated.
```

The cause seems to be that the partial view field for this macro was blank, deleting the macro helped to move the migration on.

---
_This item has been added to our backlog [AB#2236](https://umbraco.visualstudio.com/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/2236)_

**Comment by warrenbuckley** [comment]
Umbraco Version: 7.15.2
SQL Server (not CE DB)

![image](https://user-images.githubusercontent.com/1389894/63775551-7819ee80-c8d7-11e9-9f3c-6d901d111ed5.png)

**Comment by warrenbuckley** [comment]
@nul800sebastiaan & myself are unable to repro this. So closing this

**Comment by stevemegson** [comment]
I think this can only happen if you create a macro and never save it, leaving NULL in all four columns that the migration checks for a source.

While this should be an obscure edge case, I've included a fix in PR #6212 while making some other changes to the migration.

**Comment by nul800sebastiaan** [comment]
Fixed in https://github.com/umbraco/Umbraco-CMS/pull/6212
Cherry picked for 8.1.4 507c82f12237daa01f1c52ed04961e5f89673e76
